main.go

```diff
@@ -0,0 +1,176 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"net/http"
+	"net/http/pprof"
+	"net/url"
+	"os"
+	"os/signal"
+	"runtime"
+	"syscall"
+	"time"
+
+	"github.com/observatorium/observatorium/internal"
+	"github.com/observatorium/observatorium/internal/proxy"
+	"go.uber.org/automaxprocs/maxprocs"
+
+	"github.com/go-kit/kit/log/level"
+	"github.com/oklog/run"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/route"
+	"github.com/prometheus/common/version"
+)
+
+func main() {
+	profile := os.Getenv("PROFILE") != ""
+
+	debug := os.Getenv("DEBUG") != ""
+	if debug {
+		runtime.SetMutexProfileFraction(10)
+		runtime.SetBlockProfileRate(10)
+	}
+
+	opts := struct {
+		listen               string
+		gracePeriod          string
+		debugName            string
+		logLevel             string
+		logFormat            string
+		metricsReadEndpoint  string
+		metricsWriteEndpoint string
+	}{}
+
+	flag.StringVar(&opts.listen, "listen", ":8080", "The address on which internal server runs.")
+	flag.StringVar(&opts.gracePeriod, "grace-period", "5s", "The time to wait after an OS interrupt received.")
+	flag.StringVar(&opts.debugName, "debug.name", "observatorium", "The Name to add as prefix to log lines.")
+	flag.StringVar(&opts.logLevel, "log.level", "info", "The log filtering level. Options: 'error', 'warn', 'info', 'debug'.")
+	flag.StringVar(&opts.logFormat, "log.format", internal.LogFormatLogfmt, "The log format to use. Options: 'logfmt', 'json'.")
+	flag.StringVar(&opts.metricsReadEndpoint, "metrics-read-endpoint", "", "The endpoint to which to make write requests for metrics.")
```

Comment:
This is what API? Query API?

Reply:
This will proxy to Query API, yes. I'm open to naming suggestions.

Reply:
In this case we need to update the help text, which currently says:

> "The endpoint to which to make write requests for metrics.“

Reply:
so this can be mistaken by `remote read` so maybe `metrics-query-endpoint` ?